soh/soh/OTRGlobals.cpp

```diff
@@ -715,23 +716,147 @@ extern "C" void OTRExtScanner() {
+            version.patch = reader->ReadInt16();
+        }
+    }
+
+    archive = nullptr;
+
+    return version;
+}
+
+// Check that a soh.otr exists and matches the version of soh running
+// Otherwise show a message and exit
+void CheckSoHOTRVersion(std::string otrPath) {
+    const char msg[] = "Please extract the soh.otr from the Ship of Harkinian download to your folder.\n\nExiting...";
+
+    if (!std::filesystem::exists(otrPath)) {
+#if not defined(__SWITCH__) && not defined(__WIIU__)
+        Extractor::ShowErrorBox("soh.otr file is missing", msg);
+        exit(1);
+#elif defined(__SWITCH__)
+        LUS::Switch::PrintErrorMessageToScreen("\x1b[2;2HYou are missing the soh.otr file."
+                                               "\x1b[4;2HPlease re-extract it from the download.");
+#elif defined(__WIIU__)
+        OSFatal(msg);
+        exit(1);
+#endif
+    }
+
+    OTRVersion otrVersion = ReadSohVersionFromOTR(otrPath);
+
+    if (otrVersion.major != gBuildVersionMajor || otrVersion.minor != gBuildVersionMinor || otrVersion.patch != gBuildVersionPatch) {
+#if not defined(__SWITCH__) && not defined(__WIIU__)
+        Extractor::ShowErrorBox("soh.otr file version does not match", msg);
+        exit(1);
+#elif defined(__SWITCH__)
+        LUS::Switch::PrintErrorMessageToScreen("\x1b[2;2HYou have an old soh.otr file."
+                                               "\x1b[4;2HPlease re-extract it from the download.");
+#elif defined(__WIIU__)
+        OSFatal(msg);
+        exit(1);
+#endif
+    }
+}
+
+// Checks the program version stored in the otr and compares the major value to soh
+// For Windows/Mac/Linux if the version doesn't match, offer to 
+void DetectOTRVersion(std::string fileName, bool isMQ) {
+    bool isOtrOld = false;
+    std::string otrPath = LUS::Context::LocateFileAcrossAppDirs(fileName, appShortName);
+
+    // Doesn't exist so nothing to do here
+    if (!std::filesystem::exists(otrPath)) {
+        return;
+    }
+
+    OTRVersion otrVersion = ReadSohVersionFromOTR(otrPath);
+
+    if (otrVersion.major != gBuildVersionMajor) {
+        isOtrOld = true;
+    }
+
+    if (isOtrOld) {
+#if not defined(__SWITCH__) && not defined(__WIIU__)
+        char msgBuf[250];
+        char version[18]; // 5 digits for int16_max (x3) + separators + terminator
+
+        if (otrVersion.major != 0 || otrVersion.minor != 0 || otrVersion.patch != 0) {
+            snprintf(version, 18, "%d.%d.%d", otrVersion.major, otrVersion.minor, otrVersion.patch);
+        } else {
+            snprintf(version, 18, "no version found");
+        }
+
+        snprintf(msgBuf, 250,
+            "The %s file was generated with a different version of Ship of Harkinian.\nOTR version: %s\n\n"
+            "It is recommended to generate a new file to avoid issues. Would you like to regenerate it now?",
+            fileName.c_str(), version);
+
+        if (Extractor::ShowYesNoBox("Old OTR File Found", msgBuf) == IDYES) {
```

Comment:
50/50 on if we should allow users to "not" regenerate, or if we should just kill the process completely until they do.

Reply:
i'd say have the options be "regenerate/quit" and update the message to be more assertive. "this otr isn't compatable with this version of soh, you must regenerate to play."

Reply:
hmm, looks like with Windows the current implementation is limited to Yes/No (it's using windows message box instead of SDL).

I can still adjust the message to be assertive and have "No" exit the app, but I'd rather save building out an "extensible message box" as a follow PR.

Reply:
> I'd rather save building out an "extensible message box" as a follow PR.

sounds good, i figure if the text in the box makes it clear that "No" means "Quit" that'll do the trick